distributed/diagnostics/tests/test_scheduler_plugin.py

```diff
@@ -125,3 +125,16 @@ async def close(self):
 
     assert plugin.history == ["started", "closed"]
     assert plugin.scheduler is s
+
+
+@gen_cluster(client=True)
+async def test_register_scheduler_plugin(c, s, a, b):
+    temp_list = []
+
+    class Dummy(SchedulerPlugin):
+        def start(self, scheduler):
+            temp_list.append(1)
+
+    await c.register_scheduler_plugin(Dummy)
+
+    await async_wait_for(temp_list, 1)
```

Comment:
I think we can fix/simplify this test a bit by having `Dummy.start` set some attribute on the scheduler and then `assert` that the attribute is as expected after `register_scheduler_plugin` has been called. For example, something like:

```diff
diff --git a/distributed/diagnostics/tests/test_scheduler_plugin.py b/distributed/diagnostics/tests/test_scheduler_plugin.py
index 2baa914b..ef846455 100644
--- a/distributed/diagnostics/tests/test_scheduler_plugin.py
+++ b/distributed/diagnostics/tests/test_scheduler_plugin.py
@@ -129,12 +129,11 @@ async def test_lifecycle(cleanup):

 @gen_cluster(client=True)
 async def test_register_scheduler_plugin(c, s, a, b):
-    temp_list = []

     class Dummy(SchedulerPlugin):
         def start(self, scheduler):
-            temp_list.append(1)
+            scheduler.foo = "bar"

+    assert not hasattr(s, "foo")
     await c.register_scheduler_plugin(Dummy)
-
-    await async_wait_for(temp_list, 1)
+    assert s.foo == "bar"
```